lib/checkcondition.cpp

```diff
@@ -281,7 +281,7 @@ void CheckCondition::comparison()
         return;
 
     for (const Token *tok = _tokenizer->tokens(); tok; tok = tok->next()) {
-        if (Token::Match(tok, "==|!=|>")) {
+        if (Token::Match(tok, "==|!=|>|>=|<|<=")) {
```

Comment:
could you please replace this with a:

    if (!tok->isComparisonOp())
        continue;